lib/modules/dosomething/dosomething_api/resources/signup_resource.inc

```
@@ -189,7 +191,120 @@ function _signup_resource_index($user, $users, $campaigns, $count, $page, $compe
+              }
+            }
+
+            $formatted_post['kudos'] = $reactions ? $reactions : null;
+            array_push($posts, $formatted_post);
+
+            if ($post['status'] === 'rejected') {
+              $flagged = TRUE;
+            }
+          }
+        }
+
+        return [
+          'id' => $signup['signup_id'],
+          'created_at' => (new Carbon($signup['created_at']))->timestamp,
+          'campaign_run' => [
+            'id' => $signup['campaign_run_id'],
+            'current' => (dosomething_helpers_get_current_campaign_run_for_user($signup['campaign_id'], $user_object)->nid == $signup['campaign_run_id']) ? TRUE : FALSE,
+          ],
+          'uri' => null,
+          'campaign' => [
+            'id' => $signup['campaign_id'],
+            'title' => $campaign->title,
+            'campaign_runs' => $campaign->campaign_runs,
+            'language' => [
+              'language_code' => $campaign->language['language_code'],
+              'prefix' => $campaign->language['prefix'],
+            ],
+            'translations' => $campaign->translations,
+            'tagline' => $campaign->tagline,
+            'status' => $campaign->status,
+            'type' => $campaign->type,
+            'reportback_info' => $campaign->reportback_info,
+          ],
+          'reportback' => [
+            'id' => $signup['signup_id'],
+            'created_at' => (new Carbon($signup['created_at']))->timestamp,
+            'updated_at' => (new Carbon($signup['updated_at']))->timestamp,
+            'quantity' => $signup['quantity'],
+            'uri' => null,
+            'why_participated' => $signup['why_participated'],
+            'flagged' => $flagged,
+            'reportback_items' => [
+              'total' => count($signup['posts']['data']),
+              'data' => $posts,
+            ],
+          ],
+        ];
+      });
+    }
+  }
+
+  if ($transformed_response) {
```

What scenarios will `$transformed_response` not be truthy? If so, should we throw an error?

---

I think I was just doing that to determine what response to return (phoenix or rogue) but like you mentioned before, just put it under the Rogue feature flag so that should fix this up!